sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -631,6 +631,42 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
     client.dropPartitions(db, table, parts.map(lowerCasePartitionSpec), ignoreIfNotExists, purge)
   }
 
+  override def dropPartitionsByFilter(
```

Comment:
Thank you for review, @viirya and @hvanhovell .
Sure, no problem. I just thought we need to have this in `ExternalCatalog` before `Catalog Federation ( SPARK-15777 )`.
I will remove those stuff.